packages/react-instantsearch-hooks-dom/src/types/index.ts

```diff
@@ -1,5 +1,14 @@
+import type { CSSClass } from '../ui/lib/cx';
+
 /**
  * Make certain keys of an object optional
  */
 export type PartialKeys<TObj, TKeys extends keyof TObj> = Omit<TObj, TKeys> &
   Partial<Pick<TObj, TKeys>>;
+
+/**
+ * Map from CSS classes used in UI components to the `classNames` API exposed to users
+ */
+export type ClassNames<
+  TProps extends { classNames: Record<string, CSSClass> }
```

Comment:
We don't want to support all variants of `CSSClass` in the public API. We should just type it as `string`.

Reply:
yes, the output of this type is in the line below and only accepts string :)